identity-account/tests/lazy.rs

```diff
@@ -0,0 +1,151 @@
+      assert_eq!(services.len(), 2);
+
+      for service in services.iter() {
+        let service_fragment = service.as_ref().id().fragment().unwrap();
+        assert!(["my-service", "my-other-service"]
+          .iter()
+          .any(|fragment| *fragment == service_fragment));
+      }
+
+      // ===========================================================================
+      // More updates to the identity
+      // ===========================================================================
+
+      let command: Command = Command::delete_service().fragment("my-service").finish()?;
+      account.update_identity(did, command).await?;
+
+      let command: Command = Command::delete_service().fragment("my-other-service").finish()?;
+      account.update_identity(did, command).await?;
+
+      let command: Command = Command::create_method().fragment("new-method").finish()?;
+      account.update_identity(did, command).await?;
+
+      account.publish_changes(did).await?;
+
+      // ===========================================================================
+      // Second round of assertions
+      // ===========================================================================
+
+      let doc = account.resolve_identity(snapshot.identity().did().unwrap()).await?;
+      let methods = doc.methods().collect::<Vec<&IotaVerificationMethod>>();
+
+      assert_eq!(doc.service().len(), 0);
+      assert_eq!(methods.len(), 2);
+
+      for method in methods {
+        let method_fragment = method.id().fragment().unwrap();
+        assert!(["_sign-0", "new-method"]
+          .iter()
+          .any(|fragment| *fragment == method_fragment));
+      }
+
+      // ===========================================================================
+      // History assertions
+      // ===========================================================================
+
+      let client: Client = Client::from_network(network).await?;
+
+      let history: DocumentHistory = client.resolve_history(did).await?;
+
+      assert_eq!(history.integration_chain_data.len(), 1);
+      assert_eq!(history.diff_chain_data.len(), 1);
+
+      Ok(())
+    })
+  })
+  .await?;
+
+  Ok(())
+}
+
+// Repeats the test in the closure `test_runs` number of times.
+// Network problems, i.e. a ClientError trigger a re-run.
+// Other errors end the test immediately.
+async fn network_resilient_test(
+  test_runs: u32,
+  f: impl Fn(u32) -> Pin<Box<dyn Future<Output = Result<()>>>>,
+) -> Result<()> {
+  for test_run in 0..test_runs {
+    let test_attempt = f(test_run).await;
+
+    match test_attempt {
+      ok @ Ok(_) => {
+        return ok;
+      }
+      Err(AccountError::IotaError(IotaError::ClientError(client_err))) => {
+        eprintln!("test run {} errored with {:?}", test_run, client_err);
+
+        if test_run == test_runs - 1 {
+          return Err(AccountError::IotaError(IotaError::ClientError(client_err)));
+        }
+      }
```

Comment:
Nit: this branch can be simplified a bit, it changes what's printed but it doesn't matter that much either way.

```suggestion
      error @ Err(AccountError::IotaError(IotaError::ClientError(_))) => {
        eprintln!("test run {} errored with {:?}", test_run, error);

        if test_run == test_runs - 1 {
          return error;
        }
      }
```